packages/calypso-e2e/src/lib/test-account.ts

```diff
@@ -0,0 +1,161 @@
+import fs from 'fs/promises';
+import path from 'path';
+import chalk from 'chalk';
+import { BrowserContext, Page } from 'playwright';
+import { getAccountCredential, getAccountSiteURL, getCalypsoURL, config } from '../data-helper';
+import { TOTPClient } from '../totp-client';
+import { LoginPage } from './pages/login-page';
+
+/**
+ * Represents the WPCOM test account.
+ */
+export class TestAccount {
+	readonly accountName: string;
+	readonly credentials: [ string, string ];
+
+	/**
+	 * Constructs an instance of the TestAccount for the given account name.
+	 * Available test accounts should be defined in the e2e config file.
+	 */
+	constructor( accountName: string ) {
+		this.accountName = accountName;
+		this.credentials = getAccountCredential( accountName );
+	}
+
+	/**
+	 * Authenticates the account using previously saved cookies or via the login
+	 * page UI if cookies are unavailable.
+	 */
+	async authenticate( page: Page ): Promise< void > {
+		const browserContext = await page.context();
+		await browserContext.clearCookies();
+
+		if ( await this.hasFreshAuthCookies() ) {
+			this.log( 'Found fresh cookies, skipping log in' );
+			await browserContext.addCookies( await this.getAuthCookies() );
+			await page.goto( getCalypsoURL( '/' ) );
+		} else {
+			this.log( 'Logging in via Login Page' );
+			await this.logInViaLoginPage( page );
+		}
+	}
+
+	/**
+	 * Logs in via the login page UI. The verification code will be submitted
+	 * automatically if it's defined in the config file.
+	 */
+	async logInViaLoginPage( page: Page ): Promise< void > {
+		const loginPage = new LoginPage( page );
+
+		await loginPage.visit();
+		await loginPage.logInWithCredentials( ...this.credentials );
+
+		const verificationCode = this.getTOTP();
```

Comment:
Clever!

_Eventually_ I want to do a clean sweep of the config file, since it's full of cruft and unused things at this point, and establish a much more robust data structure to hold the username, password, TOTP secret and other features of the account. It would look similar to the current config file's `accounts` entry.